### PR TITLE
Users/nikita/new attestation features

### PR DESCRIPTION
**Pull request by SINIKI**

### CODE REVIEW

**Comment by SINIKI** [comment]
1 change is pending in this PR.

File: 07-Customizing-AzSK-for-your-Org/Readme.md [hunk 1]
```diff
@@ -441,6 +442,12 @@ the `Azure_Storage_DP_Encrypt_In_Transit` control.
       "ControlID": "Azure_Storage_DP_Encrypt_In_Transit",
       "Id": "AzureStorage160",
       "Recommendation": "**Note**: Use our Contoso-IT-EncryptInTransit.ps1 tool for this!"
+   },
+   {
+      "ControlID": "Azure_Storage_DP_Restrict_CORS_Access",
+      "ValidAttestationStates" : [
+        "None"
```
[the remark this answers is not in the document]
Comment (sbyna-msft) [reply 07-Customizing-AzSK-for-your-Org/Readme.md]: ok i could see the thing in the top but this json has become bigger, probably we might need to add inline comments against each example

File: 00c-Addressing-Control-Failures/Readme.md [hunk 2]
```diff
@@ -336,7 +349,13 @@ Any control with evaluation result as not passed,
 |Medium| 60|
 |Low| 90|
  
-The detailed matrix of attestation details and its expiry can be found under [this](Readme.md#how-AzSK-determines-the-effective-control-result) section.
+The detailed matrix of attestation details and its expiry can be found under [this](Readme.md#how-azsk-determines-the-effective-control-result) section. Attestation expiry date is also emitted in CSV scan result as shown in [this](../Images/02_SVT_Attest_3.PNG) image.
+
+> **Note**: All the controls are subjected to an initial grace period from the first scanned date. On expiry of grace period for a control,
+>  1. 'WillFixLater' option will be disabled for further attestation.
+>  2. attestation will expiry if the control is attested as 'WillFixLater'.
```
Comment (sbyna-msft) [inline 00c-Addressing-Control-Failures/Readme.md]: check the sentence here

File: 00c-Addressing-Control-Failures/Readme.md [hunk 3]
```diff
@@ -272,18 +277,26 @@ The columns are described as under:
 |Verify |NotAnIssue |Passed |Yes | 90 |User has ratified in the past. E.g., SQL firewall IPs scenario, where all are IPs are legitimate.|
 |Verify |WillNotFix |Exception |Yes | Based on the control severity table below|Valid security issue but a fix cannot be implemented immediately. E.g., A 'deprecated' account was found in the subscription. However, the user wants to check any dependencies before removal.|
 |Verify |WillFixLater |Remediate |Yes| Based on the control severity table below|Valid security issue but a fix cannot be implemented immediately. E.g., A 'deprecated' account was found in the subscription. However, the user wants to check any dependencies before removal.|
+|Verify |NotApplicable |Passed |Yes| 90 ||
+|Verify |StateConfirmed |Passed |Yes| Based on the control severity table below||
```
Comment (sbyna-msft) [inline 00c-Addressing-Control-Failures/Readme.md]: we need to provide some examples here

File: 00c-Addressing-Control-Failures/Readme.md [hunk 4]
```diff
@@ -189,14 +189,16 @@ the permissions required section below.
...
-The AzSK subscription and services scan cmdlets now support a new switch called *AttestControls*. When this switch is specified, 
+The AzSK subscription and services scan cmdlets now support a new switch called *ControlsToAttest*. When this switch is specified, 
 AzSK enters 'attest' mode immediately after a scan is completed. This ensures that attestation is done on the basis of the most current
 control statuses.
 
 All controls that have a technical evaluation status of anything other than 'Passed' (i.e., 'Verify' or 'Failed' or 'Manual' or 'Error') are considered 
 valid targets for attestation.
 
-To manage attestation flow effectively, 4 options are provided for the *AttestControls* switch to specify which subset of controls to target for attestation. These are described below:
+> **Note**: AzSK does not support attestation of some selected controls, and therefore fixing them is mandatory.
```
Comment (ganesh-msft) [inline 00c-Addressing-Control-Failures/Readme.md]: I think we should rather say-
'Some controls are very crucial from security stand point and hence AzSK does not support attesting them.'

File: 00c-Addressing-Control-Failures/Readme.md [hunk 5]
```diff
@@ -269,21 +274,29 @@ The columns are described as under:
 |---|---|---|---|---|---|
 |Passed |None |Passed |No | -NA- |No need for attestation. Control has passed outright!|
 |Verify |None |Verify |No | -NA- |User has to ratify based on manual examination of AzSK evaluation log. E.g., SQL DB firewall IPs list.|
-|Verify |NotAnIssue |Passed |Yes | 90 |User has ratified in the past. E.g., SQL firewall IPs scenario, where all are IPs are legitimate.|
```
Comment (SINIKI) [inline 00c-Addressing-Control-Failures/Readme.md]: sentence correction - where all IPs are legitimate.

File: 00c-Addressing-Control-Failures/Readme.md [hunk 6]
```diff
@@ -336,8 +349,15 @@ Any control with evaluation result as not passed,
...
-
+The detailed matrix of attestation details and its expiry can be found under [this](Readme.md#how-azsk-determines-the-effective-control-result) section. Attestation expiry date is also emitted in CSV scan result as shown in [this](../Images/02_SVT_Attest_3.PNG) image.
+
+> **Note**: 
+> * All the controls are subjected to an initial grace period from the first scanned date. On expiry of grace period for a control,
+>  1. 'WillFixLater' option will be disabled for further attestation.
+>  2. if the control was attested as 'WillFixLater', then attestation will expire.
+> User will then have the option to either fix the control or use other available attestation states with proper justification.
+> 
+>* Attestation may also expire before actual expiry in cases when the attested state for the control doesn't match with current control state.
```
Comment (SINIKI) [inline 00c-Addressing-Control-Failures/Readme.md]: review sentence

**Pull request merged**